src/main/webapp/app/exam/manage/exams/exam-update.component.ts

```diff
@@ -107,14 +117,75 @@ export class ExamUpdateComponent implements OnInit {
         return this.exam?.maxPoints !== undefined && this.exam?.maxPoints > 0;
     }
 
+    /**
+     * Validates the given StartDate.
+     * For RealExams, the visibleDate has to be strictly prior the startDate.
+     * For TestExams, the visible Date has to be prior or equal to the startDate.
+     */
     get isValidStartDate(): boolean {
-        return this.exam.startDate !== undefined && dayjs(this.exam.startDate).isAfter(this.exam.visibleDate);
+        if (this.exam.startDate === undefined) {
+            return false;
+        }
+        if (this.exam.testExam) {
+            return dayjs(this.exam.startDate).isSameOrAfter(this.exam.visibleDate);
+        } else {
+            return dayjs(this.exam.startDate).isAfter(this.exam.visibleDate);
+        }
     }
 
+    /**
+     * Validates the EndDate inputted by the user.
+     */
     get isValidEndDate(): boolean {
         return this.exam.endDate !== undefined && dayjs(this.exam.endDate).isAfter(this.exam.startDate);
     }
 
+    /**
+     * Calculates the WorkingTime for RealExams based on the start- and end-time.
+     */
+    get calculateWorkingTime(): number {
+        if (!this.exam.testExam) {
+            if (this.exam.startDate !== undefined && this.exam.endDate !== undefined) {
+                this.exam.workingTime = dayjs(this.exam.endDate).diff(this.exam.startDate, 's');
+            } else {
+                this.exam.workingTime = 0;
+            }
+            this.workingTimeInMinutes = this.exam.workingTime / 60;
+        }
+        return this.workingTimeInMinutes;
+    }
+
+    /**
+     * Validates the WorkingTime.
+     * For TestExams, the WorkingTime should be at least 1 and smaller / equal to the working window
+     * For RealExams, the WorkingTime is calculated based on the startDate and EndDate and should match the time difference.
+     */
+    get validateWorkingTime(): boolean {
+        if (this.exam.testExam) {
+            if (this.exam.workingTime === undefined || this.exam.workingTime < 1) {
+                return false;
+            }
+            if (this.exam.startDate !== undefined && this.exam.endDate) {
```

Comment:
```suggestion
            if (this.exam.startDate && this.exam.endDate) {
```
1) Please be consistent. Don't use `!== undefined` on the one side but not on the other. 
2) It is preferred to omit the `!== undefined` in JavaScript. So please remove this where you added it (in this file and possible others)